exporter/tanzuobservabilityexporter/metrics.go

```diff
@@ -610,3 +614,76 @@ func (regularHistogramConsumerSpec) AsHistogram(metric pdata.Metric) histogramMe
 		HistogramDataPointSlice: aHistogram.DataPoints(),
 	}
 }
+
+type exponentialHistogramDataPoint struct {
+	pdata.ExponentialHistogramDataPoint
+	bucketCounts   []uint64
+	explicitBounds []float64
+}
+
+func newExponentialHistogramDataPoint(dataPoint pdata.ExponentialHistogramDataPoint) histogramDataPoint {
```

Comment:
This function is pretty long and I find myself getting lost. If there's a simple way to break out some submethods to structure it a bit I think that would help keep it readable

Reply:
Thank you for pointing out that this code is confusing.  I added lots of comments explaining the nuances of exponential histograms and broke the function down into several smaller ones.  Let me know if it makes more sense.

Reply:
Very nice code :)